op-service/txmgr/txmgr.go

```diff
@@ -238,56 +212,31 @@ func (m *SimpleTxManager) Send(
 	}
 }
 
-// WaitMined blocks until the backend indicates confirmation of tx and returns
-// the tx receipt. Queries are made every queryInterval, regardless of whether
-// the backend returns an error. This method can be canceled using the passed
-// context.
-func WaitMined(
-	ctx context.Context,
-	backend ReceiptSource,
-	tx *types.Transaction,
-	queryInterval time.Duration,
-	numConfirmations uint64,
-) (*types.Receipt, error) {
-	return waitMined(log.New(), ctx, backend, tx, queryInterval, numConfirmations, nil)
-}
-
 // waitMined implements the core functionality of WaitMined, with the option to
 // pass in a SendState to record whether or not the transaction is mined.
-func waitMined(
-	l log.Logger,
-	ctx context.Context,
-	backend ReceiptSource,
-	tx *types.Transaction,
-	queryInterval time.Duration,
-	numConfirmations uint64,
-	sendState *SendState,
-) (*types.Receipt, error) {
-
-	queryTicker := time.NewTicker(queryInterval)
+func (m *SimpleTxManager) waitMined(ctx context.Context, tx *types.Transaction, sendState *SendState) (*types.Receipt, error) {
```

Comment:
This is probably desired to have as part of multiple implementations of the `TxManager` but since we don't have anything like ` MultiTxManager` right now this is fine

Reply:
We might need to sync offline about this, but I actually think we probably don't want multiple transaction managers & that waiting for a receipt is actually pretty core to how a transaction manager works.